src/server/acl/user.cc

```diff
@@ -236,6 +244,20 @@ void User::SetKeyGlobs(std::vector<UpdateKey> keys) {
   }
 }
 
+void User::SetPubSub(std::vector<UpdatePubSub> pub_sub) {
+  for (auto& pattern : pub_sub) {
+    if (pattern.all_channels) {
+      pub_sub_.globs.clear();
+      pub_sub_.all_channels = true;
+    } else if (pattern.reset_channels) {
+      pub_sub_.globs.clear();
+      pub_sub_.all_channels = false;
```

Comment:
I'm not sure how this works in redis and don't understand from the docs

> &<pattern>: (Available in Redis 6.2 and later) Add a glob style pattern of Pub/Sub channels that can be accessed by the user. It is possible to specify multiple channel patterns. Note that pattern matching is done only for channels mentioned by [PUBLISH](https://redis.io/commands/publish) and [SUBSCRIBE](https://redis.io/commands/subscribe), whereas [PSUBSCRIBE](https://redis.io/commands/psubscribe) requires a literal match between its channel patterns and those allowed for user.

but if you look at channel_store.h, you'll see that we keep track of all subscriber connections with references. So, once you allowed a client/user to subscribe to a channel, there's no way to stop him from reading. Moreover, if you seize his acl rights for some channel or pattern, he won't be allowed to unsubscribe, leaving him in a state where he's forced to receive messages he's not allowed to read 😆

Reply:
> Moreover, if you seize his acl rights for some channel or pattern, he won't be allowed to unsubscribe, leaving him in a state where he's forced to receive messages he's not allowed to read

Not exactly. So consider this case for example:

```
acl setuser kostas +subscribe +unsubscribe &* on >tmp
auth kostas tmp
subscribe c1
```
Then from another client:

```
acl setuser kostas -unsubscribe -subscribe
// Here the other client (kostas) is still reading the channel
publish c1 random_message -> (kostas receives this)
```

That's the chore on Valkey/Redis.

Now the interesting case is `resetchannels` which for that it completely unsubscribes the client. So I will cover this case which we don't

Reply:
See my other comment + I added a conformance test.